code/datums/spells/demon_charge.dm

```diff
@@ -0,0 +1,94 @@
+/obj/effect/immovablerod/demon
+	name = "Demon"
+	desc = "Oh no!"
+	icon = 'icons/mob/mob.dmi'
+	icon_state = "daemon"
+	notify = FALSE
+	var/mob/living/caster
+	var/max_distance = 7
+	var/turf/start_turf
+
+/datum/spell/demon_charge
+	name = "Demon Charge"
+	desc = "Perform a powerful charge, crushing all in your path."
+	clothes_req = FALSE
+	human_req = FALSE
+	base_cooldown = 5 SECONDS
+	cooldown_min = 5 SECONDS
+	action_icon_state = "immrod"
+	centcom_cancast = FALSE
+	///Used when casting, eating a mob with a mind resets this to active again
+	var/bloodcharge = TRUE
+
+/datum/spell/demon_charge/create_new_targeting()
+	return new /datum/spell_targeting/self
+
+/datum/spell/demon_charge/cast(list/targets, mob/user = usr)
+	if(get_turf(user) != user.loc)
+		to_chat(user, "<span class='warning'>You must be phased in to charge!</span>")
+		revert_cast()
+		return FALSE
+	if(!bloodcharge)
+		to_chat(user, "<span class='warning'>You need to consume a sentient new victim!</span>")
```

Comment:
Maybe say that they're lacking energy to charge and need to eat a new sentient victim. That way they get better feedback about their ability instead of just being told what they need to do.